packages/node_modules/@webex/internal-plugin-conversation/src/share-activity.js

```diff
@@ -87,6 +89,60 @@ const ShareActivity = WebexPlugin.extend({
     }
   },
 
+  /**
+   * Adds an additional GIF to the share activity
+   * Different from regular add to skip uploading to webex files service
+   * @param {File} gif
+   * @param {Object} options
+   * @param {Object} options.actions
+   * @returns {Promise}
+   */
+  addGif(gif, options) {
+    let gifToAdd = this.gifs.get(gif);
+
+    gifToAdd = Object.assign({
+      displayName: gif.name,
+      fileSize: gif.size || gif.byteLength || gif.length,
+      mimeType: gif.type,
+      url: 'https://giphy.com',
+      objectType: 'file',
+      [FILE_SYMBOL]: gif
+    }, pick(options, 'actions'));
+
+    this.uploads.set(gif, gifToAdd);
+
+    return detectFileType(gif, this.logger)
+      .then((type) => {
+        gifToAdd.mimeType = type;
+
+        return processImage({
+          file: gif,
+          type,
+          thumbnailMaxWidth: this.config.thumbnailMaxWidth,
+          thumbnailMaxHeight: this.config.thumbnailMaxHeight,
+          enableThumbnails: this.enableThumbnails,
+          logger: this.logger
+        });
+      })
+      .then((imageData) => SCR.create().then((scr) => {
```

Comment:
I would prefer if we use [promise chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#Chaining) instead of nesting. This makes the code not only more readable but also more testable.

This promise nesting can be avoided if the order of operations is changed a bit (I don't see how order matters in this case, but correct me if I'm wrong):
* create the secure content first
* Set the content location/update gif source
* Then call `processImage` to get the thumbnail

Code would look something like:
```js
return SRC.create()
.then((src) =>
  // Do what you do to set location
  return {} // Object that processImage expects
})
.then((file) => processImage(file))
.then((imageData) => {
 // Do the rest of the image data code
});

Reply:
👍